### ISSUE TITLE
Gym failing to generate IPA when touching dSYM - exit status 65 and no error logs

### ISSUE THREAD

**Issue by cmartinhas** [fastlane/fastlane#6456]
### New Issue Checklist

- [X] Updated fastlane to the latest version
- [X] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description
When running gym, I am getting "Exit status: 65" and no error reporting after it gets to the "Touching xxx.app.dSYM" step. This is the same issue others have reported in issue 1005 (https://github.com/fastlane/fastlane/issues/1005). I am running using the legacy build flag as recommended in that issue and have verified that I have dwarf w/ dysm set in the project. 

Other notes: it is generating the Archive successfully but fails to generate the IPA. If I open XCode, go to the Archives organizer, and choose to export the archive created by gym into an IPA, it creates the IPA successfully so it is only failing on the IPA creation in gym (though without reporting any actual errors). 

##### Complete output when running fastlane, including the stack trace and command used

Command being run via Terminal:
```
gym -n XXX_iOS -i "iPhone Distribution: XXX (XXXXX)" --use_legacy_build_api true
```

Terminal output: 
```
▸ Generating 'XXX.app.dSYM'
▸ Running script 'Run Script'
▸ Touching XXX.app
▸ Signing /Users/jenkins/Library/Developer/Xcode/DerivedData/Unity-iPhone-ZZZ/Build/Intermediates/ArchiveIntermediates/Unity-iPhone/InstallationBuildProductsLocation/Applications/XXX.app
▸ Touching XXX.app.dSYM
** ARCHIVE FAILED **
[10:11:17]: Exit status: 65

[!] Error building the application - see the log above
```

Log output: 
```
Touch /Users/jenkins/Library/Developer/Xcode/DerivedData/Unity-iPhone-ZZZ/Build/Intermediates/ArchiveIntermediates/Unity-iPhone/BuildProductsPath/Release-iphoneos/XXX.app.dSYM
    cd /Users/jenkins/workspace/YYY/XXX/XXX_build/Unity/Build/Xcode_iOS
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin"
    /usr/bin/touch -c /Users/jenkins/Library/Developer/Xcode/DerivedData/Unity-iPhone-ZZZ/Build/Intermediates/ArchiveIntermediates/Unity-iPhone/BuildProductsPath/Release-iphoneos/XXX.app.dSYM
```

### Configuration Files

No fastfile being used for this

### Environment

gym version: 1.11.0

Do you use bundler to execute fastlane (i.e. `bundle exec fastlane`)?
no

Do you use a Ruby environment manager (e.g. `chruby`, `rbenv`, `rvm`)?
no

**Comment by zhonghuiwen** [comment]
I was in the same position as you were, except i remembered seeing an error with a couple of my pngs and pngcrush while archiving manually through the XCode app. At first i ignored it because after archiving it through xcode i could still send it to the appstore. Turned out that was a mistake and i replaced the faulty pngs and volia!

**Comment by TKBurner** [comment]
@cmartinhas Hope you are doing well! Are you still seeing this issue? What happens if you delete that dSYM file and bump your build number, do you still run into this? Thanks!

**Comment by cmartinhas** [comment]
I was able to figure out what the issue was. The situation @zhonghuiwen described (thank you!!) got me to dig deeper into the XCode GUI and even though I had ignorable png call outs in there that I was able to archive with, it turns out that was the problem for the command line build. 

The route cause for me was Unity was compiling Android plugin png assets into the XCode project by default and the xcodebuild command line that gym runs is unfortunately not good on logging that as an issue despite the fact that it fails while the XCode GUI does not (definitely seems like a logging bug in xcodebuild itself). Once I manually told Unity to not include those files in the XCode project, it resolved the failure. Thanks!
